sql/src/test/java/org/apache/druid/sql/calcite/SqlTestFrameworkConfig.java

```diff
@@ -56,13 +64,41 @@
   /**
    * @see {@link SqlTestFrameworkConfig}
    */
-  class Rule implements AfterAllCallback, BeforeEachCallback
+  class Rule implements AfterAllCallback, BeforeEachCallback, BeforeAllCallback
   {
     Map<SqlTestFrameworkConfig, ConfigurationInstance> configMap = new HashMap<>();
     private SqlTestFrameworkConfig config;
-    private QueryComponentSupplier testHost;
+    private Function<File, QueryComponentSupplier> testHostSupplier;
     private Method method;
 
+    @Override
+    public void beforeAll(ExtensionContext context) throws Exception
+    {
+      Class<?> testClass = context.getTestClass().get();
+      SqlTestFramework.SqlTestFrameWorkModule moduleAnnotation = getModuleAnnotationFor(testClass);
```

Comment:
doubt : could there be a way in future to build the test framework without annotations as well? because while it does work very cleanly till now, I'm not sure how manageable it will be if more things keep getting added to the config. 
Maybe annotations is the only way to deal with this, but I'm not sure.

Reply:
note: this `SqlTestFramework.SqlTestFrameWorkModule` annotation class will be removed in 2 patches - but it was necessary to make this patch a standalone change

in a followup the `QueryComponentSupplier` will be part of the `SqlTestFrameWorkConfig` and around the same time there will be new ways to do it as well: most importantly it will be possible to specify it in the [URI](https://github.com/apache/druid/pull/16249) as well - so quidem tests could decide which one they want to use